### ISSUE TITLE
[Workspace Feeds] Skip the assignee step in the card creation flow from member page 

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#47308]
Par to of the [workspace feeds](https://docs.google.com/document/d/1sTg-e11jdOmregOFBhdO8ALlykKl8PqTeTEAgEMS6mU/edit#heading=h.18ccrwh58xge) project.

When admin issues a card to a policy member from the member profile page, we should skip the card assignee step as its clear we are creating the card for the member whose details we have been looking at.

> add logic to skip Assignee step when the flow is started from the member's profile page

**Comment by koko57** [comment]
I'm starting working on this issue

**Comment by koko57** [comment]
@mountiny I'm not sure why I missed it, maybe the PR for that was still open back then when I requested creating this ticket, but this feature of skipping the first step is working now 🙈 😅. 
We can close this issue or change the title, because I found a few other problems:

1) We're showing all the cards assigned to the workspace in the member's details - I think we should show only the cards that are assigned to this particular user
 
<img width="1511" alt="Screenshot 2024-08-21 at 10 26 05" src="https://github.com/user-attachments/assets/e5dff894-8336-49ea-8d24-2778136f7796">

2) When starting the issue card flow right after choosing a bank account, after we issue the card we're redirected back to the Choose Account page. We should be redirected back to Expensify Card page no matter if we start from the Choose Account page or from Expensify Card -> Issue New Card

3) On a fresh account (with no bank account) when we enable Expensify Card and the Empty state is shown we're having "Finish setup" button instead of "Issue New Card", because we're checking for `reimbursementAccountStatus !== CONST.BANK_ACCOUNT.STATE.OPEN` and we don't have `reimbursementAccountStatus` yet, so it's true and we're shown improper text - we want to have Issue New Card, right?

@mountiny so I'm wainitng for your confirmation here 🙂

**Comment by koko57** [comment]
@mountiny Vicky and me also have found an issue with Card List - after issuing card for a workspace that doesn't have any card yet the card list does not appear. I'm getting an error that an object cannot be merged with an array in Onyx. In Onyx we have an empty array instead of an object in the cards_<workspaceAccountID>_Expensify Card. We get this empty array as a response from OpenPolicyExpensifyCardsPage when we call it the first time. It is fixed after I log out/log in.

<img width="1510" alt="Screenshot 2024-08-21 at 16 40 07" src="https://github.com/user-attachments/assets/39883cfc-bdb9-4549-a98f-affc32cb2cc7">
<img width="352" alt="Screenshot 2024-08-21 at 16 40 13" src="https://github.com/user-attachments/assets/7d4f53df-14a1-4866-b26f-3d145c4bb4a3">

cc @MariaHCD

**Comment by mountiny** [comment]
@koko57 Yeah lets handle the first issues in here, I will update the OP

For the second issue, I know why its happening Auth sends empty object but PHP changes it to associative array

**Comment by koko57** [comment]
@mountiny I've found another problem and I wonder if we should fix it on the FE or the BE - after logging out and logging in again if you go straight to the Workspace -> Members and go into any member details the cards are not displayed. We get them only when going to Expensify Card page as `OpenPolicyExpensifyCardsPage` response. 

<img width="1510" alt="Screenshot 2024-08-22 at 11 16 19" src="https://github.com/user-attachments/assets/ed2856f2-a7da-4c69-a853-202c1b815da1">

So either we should call `OpenPolicyExpensifyCardsPage` when opening the member's details or members' page or the cards should be returned in `OpenWorkspaceMembersPage` (or in another call on login).

**Comment by mountiny** [comment]
Very fair, I think we can add it when you go to the `OpenWorkspaceMembersPage`, I will handle that too

**Comment by mountiny** [comment]
one web pr is up to make sure the cards are returned as empty object

**Comment by mountiny** [comment]
Added one more clean up:

4) When editing card name and card limit offline the new value isn't greyed out. I think we need to adjust it. The remaining limit also be updated wrong in offline

https://github.com/user-attachments/assets/8d034c90-8b33-47cb-aaa8-aae5bbd24550

cc @VickyStash

**Comment by VickyStash** [comment]
Hi, I'm Viktoryia from Callstack - expert contributor group - and I would like to work on this issue.

**Comment by mountiny** [comment]
Added another one 
> Integrate the isMonthlySettlementAllowed property that is now returned in the OpenPolicyExpensifyCards as part of the card settings isMonthlySettlementAllowed

**Comment by koko57** [comment]
@mountiny the first one and the third one - the 2nd was actually broken by fixing https://github.com/Expensify/App/issues/46305 (we need another solution, because this one is contradictory to what we want)

Just to make sure: after we choose account and we start the flow we want to go back to the Choose Account Page or back to Expensify Card Page?

**Comment by mountiny** [comment]
@koko57 I am not sure if it is contradictory; I think it's different going back and forth. 

If you choose the settlement bank account and start the card creation flow, then if you go back you just go back to pages based on browser history.

If you complete the flow and card is created you should go to the Expensify cards page in this case (if you create the card from the workspace member details page you go back to that page)

**Comment by koko57** [comment]
yes, but I mean contradictory to the second point:
>When starting the issue card flow right after choosing a bank account, after we issue the card we're redirected back to the Choose Account page. We should be redirected back to Expensify Card page no matter if we start from the Choose Account page or from Expensify Card -> Issue New Card

either we go back to the Choose Account or we're going back to Expensify Card page and as the bank account is chosen we don't show this page again

**Comment by mountiny** [comment]
I see what you mean. It's confusing because the Choose bank account step is essentially the first step in the create card flow in that case, so we do not go back to it once the card is created. That is because its a one-time action in this case as you mentioned

Sorry if I should have caught this earlier and confused you

**Comment by VickyStash** [comment]
> When editing card name and card limit offline the new value isn't greyed out. I think we need to adjust it. The remaining limit also be updated wrong in offline

@DylanDylann @mountiny The [PR](https://github.com/Expensify/App/pull/48260) for this issue is ready for the review

**Comment by koko57** [comment]
@mountiny while working on uncommenting the RequestExpensifyCardLimitIncrease I've encountered another problem - we don't get the bankAccountList after we login - so bankAccountList for Expensify Card page will be not available when we go straight to the Expensify Card page after logging in. Maybe it also could be returned from OpenPolicyExpensifyCardPage?

**Comment by mountiny** [comment]
Yeah we can return that? What is the bug in steps we encounter now without the bank account list? So we can test the fix out later @koko57

**Comment by koko57** [comment]
when you go to Expensify Card Page and you have at least one card issued in the remaining limit modal you will see the button that should be displayed only when the settlement bank account was NOT set up with plaid. As the bank account list is undefined the condition will also be false and the button will be displayed. After going to Wallet and find back to Expensify Card Page we have bankAccountList and the button in the remaining limit modal won't be displayed (if ofc the account was set up with plaid)
<img width="946" alt="Screenshot 2024-08-30 at 15 45 49" src="https://github.com/user-attachments/assets/052e3ab1-88e1-4729-93c6-ae22c22cfc27">

**Comment by koko57** [comment]
@mountiny for the CARDISSUE and other action types not being shown - because the message array is empty `isLegacyDeletedComment` in `isDeletedAction` is true and because of that `shouldReportActionBeVisible` returns false and that message is then filtered out.
<img width="558" alt="Screenshot 2024-08-30 at 16 19 37" src="https://github.com/user-attachments/assets/16191f87-072b-4e59-a4a5-d877f73bc5cf">

so I guess we want to have something in the message array or message.html returned from the API

**Comment by koko57** [comment]
and I've found **One more thing...** - ReconciliationAccountSettingsPage needs UpdateCardSettlementAccount being triggered. I will include this in my PR
<img width="1155" alt="Screenshot 2024-08-30 at 16 32 25" src="https://github.com/user-attachments/assets/9917bad6-dd5d-41b3-aa46-eb97f380386a">

for this one:
>When starting the issue card flow right after choosing a bank account, after we issue the card we're redirected back to the Choose Account page. We should be redirected back to Expensify Card page no matter if we start from the Choose Account page or from Expensify Card -> Issue New Card

I will use Vicky's solution with backTo param from this PR https://github.com/Expensify/App/pull/48323

**Comment by mountiny** [comment]
Thanks for the findings, I will try to include the backend fixes for this

**Comment by koko57** [comment]
> When starting the issue card flow right after choosing a bank account, after we issue the card we're redirected back to the Choose Account page. We should be redirected back to Expensify Card page no matter if we start from the Choose Account page or from Expensify Card -> Issue New Card

@mountiny I see that Vicy has already included this change in her PR,

I will include the fix for https://github.com/Expensify/App/issues/48376 in my PR with Reconciliation setting uncommented

**Comment by koko57** [comment]
PR opened https://github.com/Expensify/App/pull/48408

**Comment by mountiny** [comment]
Web PR to show the card related messages merged.

**Comment by trjExpensify** [comment]
Anything left on this one, or can we close it out?

**Comment by mountiny** [comment]
I think we can close
